### PR TITLE
Various editor fixes

### PR DESCRIPTION
**Pull request by matgis**
Breaking change: These changes require Defold `1.4.2` or later.

### User-facing changes
* Editor: Apply color to Spine gui objects in scene view.
* Editor: Saving gui files that contain Spine gui objects takes less time.

### Technical changes
* Updated `build_plugins.sh` to look for Bob under `DYNAMO_HOME` by default.
* Updated `build_plugins.sh` to not require the `tree` command, which is not available by default on macOS.

### CODE REVIEW

**Review by vlaaad — APPROVED** [review]
LGTM!

File: defold-spine/editor/src/spineguiext.clj [hunk 1]
```diff
@@ -362,17 +373,16 @@
...
-    
+  (let [node-type (:type node-desc)]
+    (cond-> (assoc node-desc
+              :size [1.0 1.0 0.0 1.0]
+              :size-mode :size-mode-auto)
+
+            (= :type-spine node-type)
+            (assoc
+              :type (:output-node-type node-type-info)
+              :custom-type (:output-custom-type node-type-info)))))
```
Comment (vlaaad) [inline defold-spine/editor/src/spineguiext.clj:384]: Nice!

File: utils/build_plugins.sh [hunk 2]
```diff
@@ -101,4 +101,10 @@ for platform in $PLATFORMS; do
     build_plugin $platform $platform_ne
 done
 
-tree $TARGET_DIR
+if command -v tree >/dev/null 2>&1; then
+    # The tree command is available. Use it.
+    tree $TARGET_DIR
```
Comment (JCash) [inline utils/build_plugins.sh:106]: It is part of the install instructions.
But sure, a fallback is probably nice.

**Review by JCash — APPROVED** [review]
LGTM, but I'm unclear which changes on `dev` warranted a new .jar + .so/.dll/.dylib?
These are only using .java and c++ files, and you haven't changed any of those?

File: defold-spine/editor/src/spineguiext.clj [hunk 3]
```diff
@@ -87,14 +88,26 @@
   (let [handle (spineext/renderable->handle renderable)
         world-transform (:world-transform renderable)
         vertex-buffer (:spine-vertex-buffer user-data)
-        vb-data-transformed (map (fn [vtx] (transform-vtx world-transform vtx)) vertex-buffer)]
+        color (:color user-data)
```
[the remark this answers is not in the document]
Comment (matgis) [reply defold-spine/editor/src/spineguiext.clj:91]: I don't know to be honest, but there was a discrepancy between the editor viewport and how it appeared in-engine so I fixed it.

**Comment by matgis** [comment]
@JCash Regarding the changed .jar + .so/.dll/.dylib, it looks like @Jhonnyg didn't commit new versions of these files when updated the code to use the new stream API in this commit: https://github.com/defold/extension-spine/commit/85f16b0e04104a2e3ef85513a1fc8ca0ea2f3419

Presumably we want the updated versions of these files in `main`?

Let me know otherwise and I'll strip them from this PR.

**Comment by JCash** [comment]
No need to strip them. Compiling with the latest sdk should always be safe.
Also, if they were missing some feature, then it's probably good.

**Pull request merged**